src/main/java/org/elasticsearch/index/query/CommonTermsQueryParser.java

```diff
@@ -160,64 +141,31 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         }
-        String field;
-        FieldMapper mapper = parseContext.fieldMapper(fieldName);
-        if (mapper != null) {
-            field = mapper.names().indexName();
-        } else {
-            field = fieldName;
+        CommonTermsQueryBuilder commonTermsQuery = new CommonTermsQueryBuilder(fieldName, value.toString())
+                .lowFreqMinimumShouldMatch(lowFreqMinimumShouldMatch)
+                .highFreqMinimumShouldMatch(highFreqMinimumShouldMatch)
+                .analyzer(analyzer);
+        
+        if (highFreqOperator != null) {
+            commonTermsQuery.highFreqOperator(highFreqOperator);
         }
-
-        Analyzer analyzer = null;
-        if (queryAnalyzer == null) {
-            if (mapper != null) {
-                analyzer = mapper.searchAnalyzer();
-            }
-            if (analyzer == null && mapper != null) {
-                analyzer = parseContext.getSearchAnalyzer(mapper);
-            }
-            if (analyzer == null) {
-                analyzer = parseContext.mapperService().searchAnalyzer();
-            }
-        } else {
-            analyzer = parseContext.mapperService().analysisService().analyzer(queryAnalyzer);
-            if (analyzer == null) {
-                throw new IllegalArgumentException("No analyzer found for [" + queryAnalyzer + "]");
-            }
+        if (lowFreqOperator != null) {
+            commonTermsQuery.lowFreqOperator(lowFreqOperator);
         }
-
-        ExtendedCommonTermsQuery commonsQuery = new ExtendedCommonTermsQuery(highFreqOccur, lowFreqOccur, maxTermFrequency, disableCoords, mapper);
-        commonsQuery.setBoost(boost);
-        Query query = parseQueryString(commonsQuery, value.toString(), field, parseContext, analyzer, lowFreqMinimumShouldMatch, highFreqMinimumShouldMatch);
-        if (queryName != null) {
-            parseContext.addNamedQuery(queryName, query);
+        if (disableCoords != null) {
+            commonTermsQuery.disableCoord(disableCoords);
         }
-        return query;
-    }
-
-
-    private final Query parseQueryString(ExtendedCommonTermsQuery query, String queryString, String field, QueryParseContext parseContext,
-            Analyzer analyzer, String lowFreqMinimumShouldMatch, String highFreqMinimumShouldMatch) throws IOException {
-        // Logic similar to QueryParser#getFieldQuery
-        int count = 0;
-        try (TokenStream source = analyzer.tokenStream(field, queryString.toString())) {
-            source.reset();
-            CharTermAttribute termAtt = source.addAttribute(CharTermAttribute.class);
-            BytesRefBuilder builder = new BytesRefBuilder();
-            while (source.incrementToken()) {
-                // UTF-8
-                builder.copyChars(termAtt);
-                query.add(new Term(field, builder.toBytesRef()));
-                count++;
-            }
+        if (cutoffFrequency != null) {
+            commonTermsQuery.cutoffFrequency(cutoffFrequency);
```

Comment:
thinking out loud, would it make sense to, instead of relying on null checks here, to initialize the variables with defaults taken from the builder, and then always set? or do we prefer how we do it with null values?